R/set_args.R

```diff
@@ -45,4 +45,112 @@ set_inherited_args <- function(fun.name, inherited.args, ..., package='gsplot'){
   # // shed non-formals
   inherited.args = function_args(package, fun.name, inherited.args, drop=TRUE)
   return(c(inherited.args, set_args(fun.name, ..., package = package)))
-}
\ No newline at end of file
+}
+
+set_legend_args <- function(object, fun.name, ...) {
+  paramsAll <- c(...)
+  
+  if(any(names(paramsAll) %in% "legend.name")) {
+  
+    names(paramsAll)[which(names(paramsAll)=='legend.name')] <- 'legend'
+    default.args <- formals(graphics::legend)
+    overall.legend <- c("x", "y", "bty", "bg", "box.lty", "box.lwd", "box.col", "cex","xjust", 
+                        "yjust", "adj", "text.width", "merge", "trace", "plot", "ncol", 
+                        "horiz", "title", "inset", "title.col", "title.adj", "xpd")  
+    not.overall <- default.args[which(!names(default.args) %in% overall.legend)]
+
+    type <- paramsAll[['type']]
+    
+    if (fun.name == "points") {
+      pt.names <- c("lwd","bg","cex")
+      names.index <- which(names(paramsAll) %in% pt.names)
+      pt.names.index <- which(pt.names %in% names(paramsAll))
+      names(paramsAll)[names.index] <- paste0("pt.", pt.names)[pt.names.index]
+      
+      if(!is.null(type) && type %in% c('l','b','o')){
+          if(is.null(paramsAll$lwd)){paramsAll$lwd <- 1}
+          if(is.null(paramsAll$lty)){paramsAll$lty <- 1}
+          if(type=='l'){paramsAll$pch <- NA}
+      } else if(!is.null(type) && type %in% c('c','h','s','S')){
+          paramsAll$pch <- NA
+          if(is.null(paramsAll$lwd)){paramsAll$lwd <- 1}
+          if(is.null(paramsAll$lty)){paramsAll$lty <- 1}
+      } else if(!is.null(type) && type %in% c('n')){
+          paramsAll$pch <- NA
+      } 
+      
+      fun.specific <- list(fill=quote(par("bg")),
+                           border=quote(par("bg")),
+                           lty=NA,
+                           lwd=NA,
+                           pch=1,
+                           density=NA,
+                           pt.bg=quote(par("bg")),
+                           pt.cex=par("cex"),
+                           pt.lwd=par("lwd"),
+                           text.col=par("col"),
+                           text.font=1)
+
+    } else if (fun.name %in% c("lines", "abline", "arrows", "segments")) {
+      
+      if(!is.null(type) && type %in% c('p', 'b','o')){
+          if(is.null(paramsAll$pch)){paramsAll$pch <- 1}
+          if(is.null(paramsAll$pt.bg)){paramsAll$pt.bg <- quote(par("bg"))}
+          if(is.null(paramsAll$pt.cex)){paramsAll$pt.cex <- par("cex")}
+          if(is.null(paramsAll$pt.lwd)){paramsAll$pt.lwd <- par("lwd")}
+          if(type=='p'){
+            paramsAll$lty <- NA
+            paramsAll$lwd <- NA
+          }
+      } else if(!is.null(type) && type %in% c('n')){
+          paramsAll$lty <- NA
+          paramsAll$lwd <- NA
+      }
+      
```

Comment:
might not need separate logic for lines and points - that is the reason that type='l' and type='p' exist